### PR TITLE
Group results

### PR DESCRIPTION
**Pull request by joshuaclayton**

### CODE REVIEW

File: app/Main.hs [hunk 1]
```diff
@@ -108,3 +112,22 @@ parseIgnorePaths = many $ strOption $
     long "ignore"
     <> metavar "PATH"
     <> help "[Allows multiple] Ignore paths that contain PATH"
+
+parseGroupings :: Parser CurrentGrouping
+parseGroupings =
+    (fromMaybe GroupByDirectory) <$> maybeGroup
```
Comment (gabebw) [inline app/Main.hs]: Are these parentheses required?

**Comment by gabebw** [comment]
Whew, there's a lot of changes! This looks reasonable to me.

**Comment by gabebw** [comment]
You may want to consider introducing a type for `(Grouping, TermMatchSet)`, since it's sprinkled over the code in a few places. Maybe `GroupedTerms`?

File: app/Main.hs (hunk 1 above)
Comment (joshuaclayton) [inline app/Main.hs]: Nope! Nice catch.

**Pull request merged**